pkg/storegateway/batch_series.go

```diff
@@ -756,13 +758,15 @@ func (s *deduplicatingBatchSet) Next() bool {
 		s.peek = nil
 	}
 	var nextEntry unloadedBatchEntry
+
 	for i := 0; i < s.batchSize; {
 		if !s.from.Next() {
 			nextBatch.Entries = nextBatch.Entries[:i+1]
 			break
 		}
 		nextEntry = s.from.At()
-		if labels.Compare(nextBatch.Entries[i].lset, nextEntry.lset) == 0 {
+
+		if labels.Equal(nextBatch.Entries[i].lset, nextEntry.lset) {
```

Comment:
Note to reviewers: `labels.Equal()` is slightly faster when you just need to compare for equality.